packages/bigtable/src/family.js

```diff
@@ -256,42 +261,51 @@ Family.prototype.getMetadata = function(callback) {
  * @resource [Garbage Collection Proto Docs]{@link https://github.com/googleapis/googleapis/blob/3592a7339da5a31a3565870989beb86e9235476e/google/bigtable/admin/table/v1/bigtable_table_data.proto#L59}
  *
  * @param {object} metadata - Metadata object.
- * @param {object|string=} metadata.rule - Garbage collection rule.
- * @param {string=} metadata.name - The updated column family name.
+ * @param {object=} metadata.rule - Garbage collection rule.
  * @param {function} callback - The callback function.
  * @param {?error} callback.err - An error returned while making this
  *     request.
  * @param {object} callback.apiResponse - The full API response.
  *
  * @example
  * family.setMetadata({
- *   name: 'updated-name',
- *   rule: 'version() > 3 || (age() > 3d && version() > 1)'
+ *   rule: {
+ *     versions: 2,
+ *     union: true
+ *   }
  * }, function(err, apiResponse) {});
  */
 Family.prototype.setMetadata = function(metadata, callback) {
+  var self = this;
+
   var grpcOpts = {
-    service: 'BigtableTableService',
-    method: 'updateColumnFamily'
+    service: 'BigtableTableAdmin',
+    method: 'modifyColumnFamilies'
   };
 
-  var reqOpts = {
-    name: this.id
+  var mod = {
+    id: this.familyName,
+    update: {}
   };
 
   if (metadata.rule) {
-    if (is.string(metadata.rule)) {
-      reqOpts.gcExpression = metadata.rule;
-    } else if (is.object(metadata.rule)) {
-      reqOpts.gcRule = Family.formatRule_(metadata.rule);
-    }
+    mod.update.gcRule = Family.formatRule_(metadata.rule);
   }
 
-  if (metadata.name) {
-    reqOpts.name = Family.formatName_(this.parent.id, metadata.name);
-  }
+  var reqOpts = {
+    name: this.parent.id,
+    modifications: [mod]
+  };
 
-  this.request(grpcOpts, reqOpts, callback);
+  this.request(grpcOpts, reqOpts, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    self.metadata = resp.columnFamilies[self.familyName];
```

Comment:
Currently untested. We check that it comes back in the callback, but not updated on the instance.